File: backend/src/main/java/com/ourride/dto/response/AuthResponse.java
package com.ourride.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {
    private String accessToken;
    private String refreshToken;
    private String tokenType = "Bearer";
    private Long expiresIn; // in seconds
    private UserInfo user;
    private boolean profileCompleted;
    private boolean requiresProfileCompletion;
    
    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class UserInfo {
        private Long id;
        private String phoneNumber;
        private String email;
        private String fullName;
        private boolean phoneVerified;
        private boolean emailVerified;
    }
}

File: mobile/client/lib/features/auth/domain/entities/auth_response.dart
import 'package:equatable/equatable.dart';
import 'user.dart';

class AuthResponse extends Equatable {
  final String accessToken;
  final String refreshToken;
  final int expiresIn;
  final User? user;
  final bool profileCompleted;
  final bool requiresProfileCompletion;

  const AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.expiresIn,
    this.user,
    required this.profileCompleted,
    required this.requiresProfileCompletion,
  });

  @override
  List<Object?> get props => [
        accessToken,
        refreshToken,
        expiresIn,
        user,
        profileCompleted,
        requiresProfileCompletion,
      ];
}

